src/picosvg/svg_reuse.py

```diff
@@ -70,3 +90,214 @@ def affine_between(
         return Affine2D.identity().translate(dx, dy)
 
     return None
+
+
+def _vectors(path: SVGPath) -> Generator[Vector, None, None]:
+    for cmd, args in path:
+        x_coord_idxs, y_coord_idxs = svg_meta.cmd_coords(cmd)
+        if cmd.lower() == "z":
+            yield Vector(0.0, 0.0)
+        else:
+            yield Vector(args[x_coord_idxs[-1]], args[y_coord_idxs[-1]])
+
+
+def _nth_vector(path: SVGPath, n: int) -> Vector:
+    return next(islice(_vectors(path), n, n + 1))
+
+
+def _angle(v: Vector) -> float:
+    # gives the directional angle of vector (unlike acos)
+    return atan2(v.y, v.x)
+
+
+def _affine_vec2vec(initial: Vector, target: Vector) -> Affine2D:
+    affine = Affine2D.identity()
+
+    # rotate initial to have the same angle as target (may have different magnitude)
+    angle = _angle(target) - _angle(initial)
+    affine = Affine2D.identity().rotate(angle)
+    vec = affine.map_vector(initial)
+
+    # scale to target magnitude
+    s = 0
+    if vec.norm() != 0:
+        s = target.norm() / vec.norm()
+
+    affine = Affine2D.compose_ltr((affine, Affine2D.identity().scale(s, s)))
+
+    return affine
+
+
+def _first_y(vectors: Iterable[Vector]) -> Optional[Vector]:
+    for idx, vec in enumerate(vectors):
+        if idx > 0 and abs(vec.y) > 0.1:
```

Comment:
again an absolute measure such as 0.1 may be either too big or too small depending of the "magnitude" of the coordinates relative to the viewBox/UPEM

Reply:
Now parameterized